spark_log_parser/loaders/__init__.py

```diff
@@ -46,12 +48,20 @@ class FileChunkStreamWrapper:
     Note #2 - once `chunks` is consumed, calls to read/lines will return empty byte buffers
     """
 
-    def __init__(self, chunks: Iterator[bytes], maximum_file_size: int = None):
+    def __init__(self, chunks: Iterator[bytes], maximum_file_size: int = DEFAULT_MAX_FILESIZE):
         assert chunks
         self.total_size: int = 0
         self._chunks: Iterator[bytes] = chunks
         self._maximum_allowed_size: int = maximum_file_size
-        self._trailing_data: bytes = b''
+        self._buffer: bytearray = bytearray()
+        # Almost all the return values from self.read() will read out of this value instead of self._buffer.
+        #  This is because reads from a memoryview are no-copy, whereas slices from a bytes/bytearray will copy
+        #  the data. This means that if we fully read self._buffer, we would have fully duplicated the data at
+        #  least one time. When dealing with file sizes that may be many 10s of GBs, this ends up resulting in a
+        #  ton of memory overhead
+        self._buffer_memview: memoryview | None = None
+        self._buffer_position: int = 0  # Index of how far into _buffer/_buffer_memview we have read
```

Comment:
Using a memoryview is not strictly necessary - I tested performance by returning slices directly from `self._buffer` as well, and that still performs quite well. However, using a memoryview is good for chopping off at least a few seconds for our larger files, though it still remains a small percentage of the overall runtime. if this seems too complicated, it would not be a big deal to remove.

Reply:
Definitely! Assuming an "ideal" usage of it, this would cut memory allocations from `self.read()` in half - if we used `self._buffer` directly, then a full `read()` of it (either all at once, or in chunk) would create a whole new copy.

In practice, since `read()` calls don't typically read in the whole buffer at once, all of those smaller slices would probably end up having a memory impact more than just 2x, since all of those smaller objects need to be created as well!